pkg/scheduler/framework/events.go

```diff
@@ -222,6 +227,14 @@ func extractPodSchedulingGateEliminatedChange(newPod *v1.Pod, oldPod *v1.Pod) *C
 	return nil
 }
 
+func extractPodGeneratedResourceClaimChange(newPod *v1.Pod, oldPod *v1.Pod) *ClusterEvent {
+	if len(newPod.Status.ResourceClaimStatuses) != len(oldPod.Status.ResourceClaimStatuses) {
```

Comment:
Since PodResourceClaimStatus doesn't have fields that are updated later, like state or timestamps, and entries added to PodResourceClaimStatus are never removed, does that mean we only need to compare the length of the slice? I saw the comment in another part of the code that caught my eye.
https://github.com/kubernetes/kubernetes/blob/52095a8b7b9b75d67a3882a21a6647e4f90ade48/plugin/pkg/auth/authorizer/node/graph_populator.go#L114-L116

Reply:
If we want to be pedantic, then yes, we would have to do a deep equal. As I decided to be pedantic elsewhere, let's to the same here :grin: 

Thanks for reminding me.

Reply:
Can you leave the similar comment about why we only need check len?

Reply:
I'm now using a common helper function to check not only the length, but also the content.